### PR TITLE
Created Hotel Booking App

### PR DESCRIPTION
**Pull request by sebalp100**
# Created Hotel Bookings app

For this project, we created a Hotel bookings app that allows users to search a list of rooms available and make reservations for 24hs.

The following was implemented:

- The user logs in to the website, only by typing the username.

loginpage

- In the navigation panel, the user can see links to: 
1. Main page
2. "Reserve" form.
3. "My reservations".
4. "Add room".
5. "Delete room".
- On the main page, the user can see a list of rooms.
homepage

- When the user selects a specific item, they can see the details with its full description.
- In the details page, the user can click the "Reserve" button.
detailspage

- When the user clicks the "Add room" link in the navigation panel they can see a form for adding a new item.
addroompage

- Made the app responsive, creating both mobile and desktop versions.
- When the user clicks the "Delete room" link in the navigation panel they can see a list of all items with title and "Delete" button.
- When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
deleteroompage

- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
- The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.
reservepage

- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).
- Added full documentation for our API.

## NOTE:
If during the start of the front-end server, the port number gives an error, replace `"start": "set PORT=3001 && react-scripts start",` for `"start": "PORT=3001  react-scripts start",` in line 31 of package.json

### CODE REVIEW

File: src/components/Reservations.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { useCurrentUserQuery } from '../api/authLog';
+import {
+  useDeleteReservationMutation,
+  useGetReservationsQuery,
+} from '../api/reservationsData';
+import './css/reservation.css';
+import Sidebar from './Sidebar';
+import MobileMenu from './MobileMenu';
+
+function Reservations() {
```
Comment (Meltrust) [inline src/components/Reservations.jsx]: - [x] When I add a reservation like this (without the city), it works and the db is populated with a reservation with no city:

image

As a result, in "My reservations" the booking appears with no city:

image

Also, the database table field is empty:

image


Kindly, add some validation so your users are not permitted to leave the "City" field form blank.

That way, the UX will be improved, and db fields won't be empty. 💪

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import './App.css';
+import AddRoom from './pages/AddRoom';
+import DeleteRoom from './components/DeleteRoom';
+import Current from './components/Current';
+import Reservations from './components/Reservations';
+import RoomDetails from './components/Detail';
+import SignUp from './components/SignUp';
+import Reservation from './pages/Reservation';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js]: - [x]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

image

That way, your code will comply with the latest standards.  💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @sebalp100,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 
-  Nice mobile implementation ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by sebalp100** [comment]
Changes requested implemented 👍 .

**Review by Meltrust — APPROVED** [review]
#### Hi @sebalp100,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Validations have been created for the reservations✔️ 
- All functions are now ES6✔️ 
- Beautiful frontend✔️
- Wonderful job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**